File: backend/app/engines/premium_discount/PremiumDiscountResult.py
from __future__ import annotations

from typing import Literal
from typing import TypedDict


class PremiumDiscountResult(TypedDict):
    valid: bool
    zone: Literal["PREMIUM", "DISCOUNT", "EQUILIBRIUM", "UNKNOWN"]
    currentPrice: float
    rangeHigh: float
    rangeLow: float
    equilibrium: float
    distanceToEquilibrium: float
    inPremium: bool
    inDiscount: bool
    inEquilibrium: bool
    biasAligned: bool
    passedRules: list[str]
    failedRules: list[str]
    reasons: list[str]
    warnings: list[str]

File: frontend/src/models/PremiumDiscountResult.ts
export type PremiumDiscountZone = "PREMIUM" | "DISCOUNT" | "EQUILIBRIUM" | "UNKNOWN";

export interface PremiumDiscountResult {
  valid: boolean;
  zone: PremiumDiscountZone;
  currentPrice: number;
  rangeHigh: number;
  rangeLow: number;
  equilibrium: number;
  distanceToEquilibrium: number;
  inPremium: boolean;
  inDiscount: boolean;
  inEquilibrium: boolean;
  biasAligned: boolean;
  passedRules: string[];
  failedRules: string[];
  reasons: string[];
  warnings: string[];
}
